x-pack/plugins/maps/public/shared/layers/sources/es_search_source/create_source_editor.js

```diff
@@ -68,6 +96,11 @@ export class CreateSourceEditor extends Component {
       return;
     }
 
+    // Turn off filterByMapBounds when index contains a limited number of documents
+    const indexDocCount = await this.loadIndexDocCount(indexPattern.title);
```

Comment:
I'd remove the `doc_count_unset` indirection and would just put the try-catch around `loadIndexDocCount`, it will remove one magic variable.